src/payment-request.ts

```diff
@@ -54,27 +79,69 @@ export async function createPaymentRequest(
           }
         }
       `,
-      { sessionId, payeeName, payeeEmail, sessionPreviewData }
+      {
+        sessionId,
+        payeeName,
+        payeeEmail,
+        sessionPreviewData: validatedSessionData.sessionPreviewData,
+        //TODO amount: validatedSessionData.amount
+      }
     );
   return response?.insert_payment_requests_one;
 }
 
-export function buildSessionPreviewData(
-  session: Session,
-  sessionPreviewKeys: KeyPath[]
-): PaymentRequest["sessionPreviewData"] {
+// lookup the "amount" passport key from the pay component of a given flow
+async function getPaymentAmountKey(
+  client: GraphQLClient,
+  flowId: string
+): Promise<string> {
+  const flowGraph = await getLatestFlowGraph(client, flowId);
+  if (!flowGraph) {
+    throw new Error("flow not found");
+  }
+  // search for pay component from the bottom of the root node upwards
+  let payNode: Node | undefined;
+  flowGraph._root.edges.reverse().forEach((rootNodeId) => {
+    const node = flowGraph[rootNodeId];
+    if (node.type === ComponentType.Pay) {
+      payNode = node;
+      return;
+    }
+  });
+  if (!payNode) {
+    throw new Error("flow must contain a pay node");
+  }
+  return (payNode.data?.fn as string) || "amount";
+}
```

Comment:
is it worth adding a comment here that this is explicitly assuming one Pay node per flow and the passport key is reflecting the _first_ Pay node it finds from bottom up? (which is a totally fair assumption for now based on prod content, but nice to be explicit about in my opinion!)

Reply:
As mentioned above, this now uses breadcrumbs to find the relevant pay component.